agent/consul/rpc.go

```diff
@@ -293,7 +293,7 @@ func (s *Server) handleNativeTLS(conn net.Conn) {
 		s.handleSnapshotConn(tlsConn)
 
 	case pool.ALPN_RPCGRPC:
-		s.grpcHandler.Handle(conn)
+		s.grpcHandler.Handle(tlsConn)
```

Comment:
This was an existing bug likely due to a simple copypaste gone awry. All of the other branches take a `tlsConn` instead of a `conn` to be on the correct side of the multiplex envelope.